### PR TITLE
Added support S3DiskStorage for file upload/download

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/1862

### CODE REVIEW

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveFile.scala [hunk 1]
```diff
@@ -1,14 +1,62 @@
...
+    val path       = intermediateFolders(storage.project, description.uuid, description.filename)
+    val key        = path.toString
+    def s3Sink     = S3.multipartUpload(storage.value.bucket, key).withAttributes(attributes)
+    IO.fromFuture(
+      S3.getObjectMetadata(storage.value.bucket, key)
+        .withAttributes(attributes)
+        .runWith(Sink.last)
+        .flatMap {
+          case None    =>
+            println("here2222")
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveFile.scala:40]: To delete

File: build.sbt [hunk 2]
```diff
@@ -498,14 +499,15 @@ lazy val storagePlugin = project
...
+      akkaHttpXml,
       alpakkaS3,
       akkaSlf4j        % Test,
       akkaTestKitTyped % Test,
       akkaHttpTestKit  % Test,
       h2               % Test,
       logback          % Test,
       scalaTest        % Test
-    ),
+    ) ++ dockerTestKit,
```
Comment (bogdanromanx) [inline build.sbt:510]: this is not in the test scope

File: .github/workflows/ci-storage-plugin.yml [hunk 3]
```diff
@@ -42,7 +42,10 @@ jobs:
       - name: Static Analysis
         run: sbt -Dsbt.color=always -Dsbt.supershell=false "project storagePlugin" clean scalafmtCheck test:scalafmtCheck scalafmtSbtCheck scapegoat
       - name: Tests
-        run: sbt -Dsbt.color=always -Dsbt.supershell=false "project storagePlugin" clean coverage test coverageReport coverageAggregate
+        run: |
+          grep -qF 'bucket.my-domain.com' /etc/hosts || echo "127.0.0.1 bucket.my-domain.com" | sudo tee -a /etc/hosts
```
Comment (bogdanromanx) [inline .github/workflows/ci-storage-plugin.yml:46]: is sudo working in gh-actions?

File: delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageAccessSpec.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    "succeed verifying the bucket" in {
+      access(iri, storage).accepted
+    }
+
+    "fail when bucket does not exist" in {
+      access(iri, storage.copy(bucket = "other")).rejectedWith[StorageNotAccessible]
+    }
+
+    "delete bucket" in {
```
Comment (imsdu) [inline delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageAccessSpec.scala:58]: afterAll

File: delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveAndFetchFileSpec.scala [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      accessKey = Some(Secret.decrypted(AccessKey)),
+      secretKey = Some(Secret.decrypted(SecretKey)),
+      region = Some(Region.EU_CENTRAL_1),
+      readPermission = read,
+      writePermission = write,
+      maxFileSize = 20,
+      Decrypted
+    )
+    val uuid         = UUID.fromString("8049ba90-7cc6-4de5-93a1-802c04200dcc")
+    val project      = ProjectRef(Label.unsafe("org"), Label.unsafe("project"))
```
Comment (imsdu) [inline delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveAndFetchFileSpec.scala:62]: There is a ProjectRef.unsafe

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageValue.scala [hunk 6]
```diff
@@ -113,6 +118,38 @@ object StorageValue {
...
+        case None                                   => region.fold(s"https://$bucket.s3.amazonaws.com")(r => s"https://$bucket.s3.$r.amazonaws.com")
+      }
+
+    /**
+      * @return these settings converted to an instance of [[akka.stream.alpakka.s3.S3Settings]]
+      */
+    def toAlpakkaSettings: s3.S3Settings = {
+      val credsProvider = (accessKey, secretKey) match {
+        case (Some(accessKey), Some(secretKey)) =>
+          StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKey.value, secretKey.value))
```
Comment (bogdanromanx) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageValue.scala:134]: can these operations fail?

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveFile.scala (hunk 1 above)
Comment (bogdanromanx) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveFile.scala:40]: println

File: delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveAndFetchFileSpec.scala [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  "An S3Storage operations" should {
+    val iri = iri"http://localhost/s3"
+
+    val storageValue = S3StorageValue(
+      default = false,
+      algorithm = DigestAlgorithm.default,
+      bucket = "bucket2",
+      endpoint = Some(s"http://$VirtualHost:$MinioServicePort"),
+      accessKey = Some(Secret.decrypted(AccessKey)),
+      secretKey = Some(Secret.decrypted(SecretKey)),
```
Comment (imsdu) [inline delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3StorageSaveAndFetchFileSpec.scala:54]: We need a test to see if we handle the error correctly invalid credentials

**Review by imsdu — CHANGES_REQUESTED** [review]

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageValue.scala (hunk 6 above)
Comment (umbreak) [reply delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageValue.scala:134]: No

**Review by imsdu — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**